### DISCUSSION TITLE
dvas performances

### DISCUSSION THREAD

**Discussion by fpavogt** [MeteoSwiss/dvas#154] [Show and tell]
Hey @MCH-MDA/dvas-aficionados, 

There's a new `dvas_optimize` entry point that allows to fine-tune one key parameter to speed things up. And I would be very curious to compare performances between our different machines. 

For those that have access to a Python 3.8/3.9 **anywhere** (MCH/private laptops, servers, cloud, etc ...), would you mind doing the following, and reporting back the results in this discussion ? Thanks !

  * 1: Clone the following specific branch to your machine:
      ```
      cd any/where/you/like/
      git clone -b develop_vof git@github.com:MCH-MDA/dvas.git dvas_codedev
      ```
  * 2: Install dvas locally
     ```
     pip install -e ./dvas_codedev
     ```
  * 3: Run the dvas initialization step
     ```
     dvas_init_arena
     ```
   * 4: Run the optimization routine, and report the outcome in this thread:
     ```
     cd dvas_proc_arena
     dvas_optimize
     ```

**Comment by fpavogt** [comment]
Where: MCH laptop (WSL - Ubuntu 20.04), Python 3.9
cpu count: 4
Performances:
```
  chunk_size: 50 <=> 47.99s
  chunk_size: 112 <=> 27.15s
  chunk_size: 175 <=> 27.36s
  chunk_size: 237 <=> 27.49s
  chunk_size: 300 <=> 32.17s
  chunk_size: 300 <=> 29.39s
  chunk_size: 237 <=> 27.99s
  chunk_size: 175 <=> 27.43s
  chunk_size: 112 <=> 27.29s
  chunk_size: 50 <=> 44.33s
```

**Reply by cfelix-ch** [reply]
There is apparently room for improvement with the next generation of MCH laptops... :)

**Comment by fpavogt** [comment]
Where: MBP 2019 (OSX 10.15.7), Python 3.8
cpu count: 16
Performances:
```
  chunk_size: 50 <=> 5.09s
  chunk_size: 112 <=> 3.73s
  chunk_size: 175 <=> 4.29s
  chunk_size: 237 <=> 4.87s
  chunk_size: 300 <=> 5.92s
  chunk_size: 300 <=> 6.00s
  chunk_size: 237 <=> 4.79s
  chunk_size: 175 <=> 4.73s
  chunk_size: 112 <=> 4.42s
  chunk_size: 50 <=> 6.49s
```
Performances (capping the number of cpus to 4 with `dvas_optimize --n-cpus 4`):
```
  chunk_size: 50 <=> 11.61s
  chunk_size: 112 <=> 7.88s
  chunk_size: 175 <=> 9.19s
  chunk_size: 237 <=> 9.55s
  chunk_size: 300 <=> 10.20s
  chunk_size: 300 <=> 10.34s
  chunk_size: 237 <=> 9.80s
  chunk_size: 175 <=> 9.57s
  chunk_size: 112 <=> 8.31s
  chunk_size: 50 <=> 12.47s
```
